runtime/src/main/java/com/datastax/oss/quarkus/runtime/reactive/Wrappers.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.quarkus.runtime.reactive;
+
+import io.smallrye.mutiny.Multi;
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import org.reactivestreams.Publisher;
+
+public class Wrappers {
+  public static <T> Multi<T> toMulti(Publisher<T> publisher) {
+    Context context = Vertx.currentContext();
+    if (context != null) {
+      return Multi.createFrom()
+          .publisher(publisher)
+          .emitOn(command -> context.runOnContext(x -> command.run()));
+    } else {
+      return Multi.createFrom().publisher(publisher);
+    }
+  }
```

Comment:
Took me sometime to understand line 29. I think it's because we don't expect the `Executor` interface to be used as a lambda (it's not marked `@FunctionalInterface`). I know it's more verbose, but how about this:

```suggestion
  public static <T> Multi<T> toMulti(Publisher<T> publisher) {
    Multi<T> multi = Multi.createFrom().publisher(publisher);
    Context context = Vertx.currentContext();
    if (context != null) {
      multi = multi.emitOn(new VertexContextExecutor(context));
    }
    return multi;
  }

  private static class VertexContextExecutor implements Executor {

    private final Context context;

    public VertexContextExecutor(Context context) {
      this.context = context;
    }

    @Override
    public void execute(@NonNull Runnable command) {
      context.runOnContext(x -> command.run());
    }
  }
```

Reply:
Maybe we could just change this:
`.emitOn(command -> context.runOnContext(x -> command.run()));`
to 
`.emitOn(executor -> context.runOnContext(x -> executor.run()));`
It should solve the problem. wdyt?